### PR TITLE
samples: Updating notes for central_smp_client readme

### PR DESCRIPTION
**Pull request by dchat-nordic**
Added notes explicitly specifying that this sample
is not intended as device programmer.

Jira: NCSDK-14095

Signed-off-by: Dominik Chat <dominik.chat@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -316,6 +316,7 @@ Documentation
 
   * Documentation for :ref:`degugging of nRF5340 <debugging>` in :ref:`working with nRF5340 DK<ug_nrf5340>` user guide.
   * Section describing how to enable Amazon Frustration-Free Setup (FFS) support in :ref:`ug_matter_configuring_device_identification`.
+  * Notes for :ref:`bluetooth_central_dfu_smp` specifying the intended use for the sample.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:319]:
```suggestion
  * Notes to the :ref:`bluetooth_central_dfu_smp` sample document specifying the intended use of the sample.
```

File: samples/bluetooth/central_smp_client/README.rst [hunk 2]
```diff
@@ -10,6 +10,10 @@ Bluetooth: Central SMP Client
 The Central Simple Management Protocol (SMP) Client sample demonstrates how to use the :ref:`dfu_smp_readme` to connect to an SMP Server and send a simple echo command.
 The response, which is received as CBOR-encoded data, is decoded and printed.
 
+.. note::
+   This sample does *not* provide the means to program the device using DFU.
+   The sample is intended to show communication between SMP Client and SMP Server.
```
Comment (peknis) [inline samples/bluetooth/central_smp_client/README.rst:15]:
```suggestion
   This sample does not provide the means to program a device using DFU.
   It demonstrates the communication between SMP Client and SMP Server.
```

File: samples/bluetooth/central_smp_client/README.rst [hunk 3]
```diff
@@ -32,6 +36,9 @@ The sample supports the following development kits:
 
 The sample also requires a device running `mcumgr`_ with transport protocol over Bluetooth® Low Energy, for example, another development kit running the :ref:`smp_svr_sample`.
 
+.. note::
+   This sample will *not* program the device using DFU.
```
Comment (peknis) [inline samples/bluetooth/central_smp_client/README.rst:40]:
```suggestion
   This sample does not program the device using DFU.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Review by grochu — APPROVED** [review]
Looks good, it should be clear now, what the sample does.

**Pull request merged**